tests/codegen/ccode/scripts/loops.py

```diff
@@ -0,0 +1,119 @@
+from pyccel.decorators import types
+
+#==============================================================================
+
+@types( int )
+def sum_natural_numbers( n ):
+    x = 0
+    for i in range( 1, n+1 ):
+        x += i
+    return x
+
+# ...
+@types( int )
+def factorial( n ):
+    x = 1
+    for i in range( 2, n+1 ):
+        x *= i
+    return x
+
+# ...
+@types( int )
+def fibonacci( n ):
+    x = 0
+    y = 1
+    for i in range( n ):
+        z = x+y
+        x = y
+        y = z
+    return x
+
+# ...
+@types( int )
+def double_loop( n ):
+    x = 0
+    for i in range( 3, 10 ):
+        x += 1
+        y  = n*x
+        for j in range( 4, 15 ):
+            z = x-y
+    return z
+
+# ...
+@types( int )
+def sum_natural_numbers_2( n ):
+    x = 0.
+    for i in range( 1, n+1 ):
+        x= x+i
+    return x
+
+# ...
+@types( int )
+def factorial_2( n ):
+    x = 1.
+    for i in range( 2, n+1 ):
+        x = x*i
+    return x
+
+# ...
+@types( int )
+def fibonacci_2( n ):
+    x = 0.
+    y = 1.
+    for i in range( n ):
+        z = x+y
+        x = y
+        y = z
+    return x
+
+# ...
+@types( int )
+def double_loop_2( n ):
+    x = 0.
+    for i in range( 3, 10 ):
+        x = x+1
+        y  = n*x
+        for j in range( 4, 15 ):
+            z = x-y
+    return z
+
+# ...
+#@types( 'int[:,:](order=C)' )
+#def double_loop_on_2d_array_C( z ):
+#
+#    from numpy import shape
+#
+#    s = shape( z )
+#    m = s[0]
+#    n = s[1]
+#
+#    for i in range( m ):
+#        for j in range( n ):
+#            z[i,j] = i-j
+
+
+# ...
+#@types( 'int[:,:](order=F)' )
+#def double_loop_on_2d_array_F( z ):
+#
+#    from numpy import shape
+#
+#    s = shape( z )
+#    m = s[0]
+#    n = s[1]
+#
+#    for i in range( m ):
+#        for j in range( n ):
+#            z[i,j] = i-j
+
+# ...
+#@types( 'real[:], real[:]' )
+#def product_loop_on_real_array( z, out ):
+#
+#    from numpy     import shape
+#
+#    s = shape( z )
+#    n = s[0]
+#
+#    for i in range(n):
+#        out[i] = z[i]**2
```

Comment:
Should we put these tests in a separate files and mark them as xfail rather than commenting them?

Reply:
arrays are not implemented at all in C that's why i prefered to comment it rather than mark them xfails

> We should add end to end tests that check the results not just tests to see if something is generated
yes this tests just tests if the codegen mechanism works
we should definitely add tests to check the results of the functions
but this is for later, the goal of this branch is to add the ccode mechanism and be sure that it works not that it's correct.
in order for us to tests the results we need to be able to compile a c program and also be able to generate a  shared library that can be imported in Python

Reply:
In order to test the results with epyccel we need to be able to compile a c program and also be able to generate a shared library that can be imported in Python.
If we just use pyccel we can add the compile command manually for basic testing. 

The test "test_imports_compile" in tests/pyccel/test_pyccel.py does this for fortran

Reply:
An array in C does exist. It is constructed as a pointer:
```
int a[4];
```
would be equivalent to
```
a = np.empty(4,dtype=int)
```

Reply:
yes we need to add compilation of a c program.

Reply:
for the arrays it's not clear to me how we can implement it, for example in Fortran we can add tow arrays but we can't in C
`c = a + b ` where `a` and `b` are arrays

Reply:
Obviously these kinds of operations would have to be done element-wise. We can create a separate issue for this as it seems to be beyond the scope of this pull request

Reply:
yes it should be solved in another pull request